sdk/formrecognizer/ai-form-recognizer/src/error.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ErrorModel, InnerError } from "./generated";
+
+/**
+ * Returns the innermost error that has a message field.
+ *
+ * @internal
+ * @param error - the error to unwrap
+ * @returns - the innermost populated error
+ */
+function unwrap(error: ErrorModel | InnerError): ErrorModel {
+  if (error.innererror !== undefined && error.innererror.message !== undefined) {
+    return unwrap(error.innererror);
+  }
+
+  return error as ErrorModel;
```

Comment:
I wonder why the recursion here, could you get an error with more than one layer of inner errors?